sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -132,7 +132,9 @@ abstract class Optimizer(sessionCatalog: SessionCatalog)
       OptimizeCodegen) ::
     Batch("RewriteSubquery", Once,
       RewritePredicateSubquery,
-      CollapseProject) :: Nil
+      CollapseProject) ::
+    Batch("UpdateAttributeReferences", Once,
```

Comment:
The original motivation is to fix nullability inconsistency in logical plans and I couldn't find concrete examples to make the optimization better by using this new rule. Any idea?